drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -370,16 +402,89 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 	}
 
 	if (i == clk_values) {
-		dev_err(bus->dev, "%s: could not find clock value for bandwidth %d\n",
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
+		slave_prop = &s_rt->slave->prop;
+
+		/*
+		 * Find an available manager lane for the first Peripheral
+		 * Testing the first Peripheral is enough because we can't use multi-lane
+		 * if we can't find any available lane for the first Peripheral
+		 */
+		for (i = 1; i < SDW_MAX_LANES; i++) {
+			if (!slave_prop->lane_maps[i])
+				continue;
+
+			required_bandwidth = 0;
+			list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+				required_bandwidth += m_rt->stream->params.rate *
+					hweight32(m_p_rt->ch_mask) *
+					m_rt->stream->params.bps;
+			}
+			if (required_bandwidth <= curr_dr_freq - bus->lane_used_bandwidth[i]) {
+				/* Check if m_lane is connected to all Peripherals */
+				if (!is_all_peripherals_connected(m_rt, slave_prop->lane_maps[i])) {
+					dev_dbg(bus->dev,
+						"some Peripherals are not connected to %d\n",
+						slave_prop->lane_maps[i]);
+					continue;
+				}
+				m_lane = slave_prop->lane_maps[i];
+				dev_dbg(&s_rt->slave->dev,
+					"M lane %d P lane %d can be used\n",
+					m_lane, i);
+				bus->lane_used_bandwidth[i] += required_bandwidth;
+				/*
+				 * Use non-zero manager lane, subtract the lane 0
+				 * bandwidth that is already calculated
+				 */
+				bus->params.bandwidth -= required_bandwidth;
+				use_multi_lane = true;
+				break;
+			}
+		}
+
+		if (!use_multi_lane) {
+			dev_err(bus->dev, "%s: could not find an available lane\n", __func__);
+			return -EINVAL;
+		}
+
+		/* Set Peripheral lanes */
+		list_for_each_entry(s_rt, &m_rt->slave_rt_list, m_rt_node) {
+			slave_prop = &s_rt->slave->prop;
+			for (i = 1; i < SDW_MAX_LANES; i++) {
+				if (slave_prop->lane_maps[i] == m_lane) {
+					dev_dbg(&s_rt->slave->dev, "Set Peripheral lane = %d\n", i);
+					list_for_each_entry(s_p_rt, &s_rt->port_list, port_node) {
+						s_p_rt->lane = i;
+						break;
+					}
+				}
+			}
+		}
+		/* Set Manager lanes */
+		/* Assume there is only one Manager in multi-lane configurations. */
```

Comment:
I don't follow this comment, in the most agressive 192kHz 8ch bandwidth, we will have two links with two lanes each.

Reply:
> I don't follow this comment, in the most agressive 192kHz 8ch bandwidth, we will have two links with two lanes each.

Hmm, I can't test the 2 links case for now. But for that case, sdw_compute_bus_params() will be called twice for each Manager, right?

Reply:
Edited the comments. Hope it will be more readable.